GVRf/Framework/src/org/gearvrf/GVRResourceVolume.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import android.os.Environment;
+
+/**
+ * Models a file system which supports stream I/O.
```

Comment:
GVRAndroidResource is used for loading textures and meshes. It should be used (and extended if necessary) to load models. If GVRAndroidResource is not good enough - we should replace it with something better that is used for textures, meshes and models and prevents the overload fan out GVRAndroidResource does.

We can't have the implementation (or worse, future implementation predictions) drive the APIs we expose. Let's come up with best API we can and then we can worry about the implementation.